object_store/src/local.rs

```diff
@@ -678,6 +618,94 @@ impl ObjectStore for LocalFileSystem {
     }
 }
 
+impl LocalFileSystem {
+
+    fn list_with_maybe_offset(&self, prefix: Option<&Path>, maybe_offset: Option<&Path>) -> BoxStream<'static, Result<ObjectMeta>> {
+
+        let config = Arc::clone(&self.config);
+
+        let root_path = match prefix {
+            Some(prefix) => match config.prefix_to_filesystem(prefix) {
+                Ok(path) => path,
+                Err(e) => return futures::future::ready(Err(e)).into_stream().boxed(),
+            },
+            None => config.root.to_file_path().unwrap(),
+        };
+
+        let walkdir = WalkDir::new(root_path)
+            // Don't include the root directory itself
+            .min_depth(1)
+            .follow_links(true);
+
+        let maybe_offset_clone = maybe_offset.map(move |o| o.clone());
```

Comment:
It's usually idiomatic rust to rebind a name in this situation, eg:
```suggestion
        let maybe_offset = maybe_offset.map(move |o| o.clone());
```

Reply:
There's also a cloned method for this:
```suggestion
        let maybe_offset = maybe_offset.cloned();
```

Reply:
@adamreeve Thanks for all the helpful comments! I have incorporated them.